CONTRIBUTING.md

```diff
@@ -1,20 +1,96 @@
 # Contributing
 
-## Opening an Issue
-The title of a Bug or Enhancement should clearly indicate what is broken or desired. Use the description to explain possible solutions or add details and (especially for Enhancemnts) explain *how* or *why* the issue is broken or desired.
+- [Reporting Issues](#reporting-issues)
+  - [Opening an Issue](#opening-an-issue)
+- [Contributing Code](#contributing-code)
+  - [Build Dependencies](#build-dependencies)
+  - [Building the SDK](#building-the-sdk)
+  - [Running Tests](#running-tests)
+  - [Git Commit Guidelines](#git-commit-guidelines)
+- [Updating the Documentation](#updating-the-documentation)
 
-### Grammar
-While quibbling about grammar in issue titles may seem a bit pedantic, adhering to some simple rules can make it much easier to understand a Bug or an Enhancement from the title alone. For example, is the title **"Browsers should support blinking text"** a bug or a feature request?
+## Reporting Issues
 
-- Enhancements: The title should be an imperative statement of how things should be. **"Add support for blinking text"**
+### Opening an Issue
+
+The title of a Bug or Enhancement should clearly indicate what is broken or desired. Use the description to
+explain possible solutions or add details and (especially for Enhancemnts) explain *how* or *why* the issue is
+broken or desired.
+
+#### Grammar
 
+While quibbling about grammar in issue titles may seem a bit pedantic, adhering to some simple rules can make it much
+easier to understand a Bug or an Enhancement from the title alone. For example, is the title **"Browsers should support
+blinking text"** a bug or a feature request?
+- Enhancements: The title should be an imperative statement of how things should be. **"Add support for blinking text"**
 - Bugs: The title should be a declarative statement of how things are. **"Text does not blink"**
 
-## Git Commit Guidelines
+## Contributing Code
+
+### Build Dependencies
+
+Before you can build the Cisco Spark JS SDK, you will need the following dependencies:
+- [Node.js](https://nodejs.org/) 6.x (LTS)
+  - We recommend using [nvm](https://github.com/creationix/nvm) (or [nvm-windows](https://github.com/coreybutler/nvm-windows))
+    to easily switch between Node.js versions
+- [Git](https://git-scm.com/)
+
+You will need to create a file called `.env` that defines, at a minimum:
+
+- `CISCOSPARK_CLIENT_ID`
+- `CISCOSPARK_CLIENT_SECRET`
+- `CISCOSPARK_REDIRECT_URI`
+- `CISCOSPARK_SCOPE`
+
+You can get these values by registering a new integration on the [Spark for Developers](https://developer.ciscospark.com/add-integration.html) portal.
+
+### Building the SDK
+
+Fork the [spark-js-sdk](https://github.com/ciscospark/spark-js-sdk/) repository and `git clone` your fork:
+
+    git clone https://github.com/your-username/spark-js-sdk.git
```

Comment:
The `git clone` command will be shown as a code block, though it's not fenced by back-ticks.